pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -69,13 +73,44 @@ type akvKMProvider struct {
 	tenantID     string
 	clientID     string
 	cloudName    string
+	resource     string
 	certificates []types.KeyVaultValue
 	keys         []types.KeyVaultValue
 	cloudEnv     *azure.Environment
-	kvClient     *kv.BaseClient
+	kvClient     kvClient
```

Comment:
Both kvClient and kv.BaseClient rely on autorest (see https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/keyvault/v7.1/keyvault#BaseClient) which is what we need to avoid as it's deprecated. I have refactored the code to replace whatever depends on autorest with azidentity (see #1904 ). 
In a nutshell, instead of "github.com/Azure/azure-sdk-for-go/services/keyvault/v7.1/keyvault", we can use azidentity with azcertificates, azsecrets, and azkeys. They abstract away the process of getting AAD access token and authenticating with AKV. Currently in my PR I have implemented the workload identity scenario (which is the scenario for AKS), but it can be extended to other auth types like managed identity and CLI using changedIdentityCredential.
@susanshi @duffney  We can refactor this PR to use azidentity. Let me know if you need this to be done on this PR and I am happy to help. Otherwise, we can merge this PR first, and then resolve the conflicts in my PR.

Reply:
thanks @shahramk64 , given this PR has been open for a while, I would recommend complete this first, we want to avoid further expansion more in scope :)

Reply:
I agree. Let's merge this PR once it's ready, I'll resolve the conflicts with my PR after that.